### PR TITLE
Fix incorrect TBB version in Java API

### LINKED ISSUE #1254: Java API failed to load TBB in oneAPI due to incorrect TBB version
Update: this issue is still effective for production version oneDAL 2021.1, 2021.2, 2021.3.
----------------------------------------------------------------------------------
**Describe the bug**
https://github.com/oneapi-src/oneDAL/blob/master/java/com/intel/daal/utils/LibUtils.java#L143
This line creates fullname as __libtbb.so.2__, however, in oneAPI SDK Beta 10, TBB lib version is actually __libtbb.so.12__ (/opt/intel/oneapi/tbb/2021.1-beta10/env/../lib/intel64/gcc4.8/libtbb.so.12), leads to libJavaAPI.so failed to load. 

$ ldd libJavaAPI.so
        linux-vdso.so.1 (0x00007ffcf95d3000)
        libtbb.so.12 => not found
        libtbbmalloc.so.2 => not found
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fd499f45000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fd499ba7000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fd49998f000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fd49959e000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fd49939a000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fd4a7f60000)

**To Reproduce**
Call any Java API in oneAPI beta 10

**Environment:**
 - OS: [e.g. Ubuntu 18.04]
 - Version: [oneAPI beta 10]

### PR DESCRIPTION
**Pull request by xwu99**
# Description
Please include a summary of the change. For large or complex changes please include enough information to introduce your change and explain motivation for it.

Changes proposed in this pull request:

closes #1254

### CODE REVIEW

File: java/com/intel/daal/utils/LibUtils.java [hunk 1]
```diff
@@ -140,7 +140,7 @@ public static String createLibraryFileName(String name) throws IOException
 
         if (OSname.startsWith("linux")) {
             if (name.contains("tbb")) {
-                fullName = "lib" + name + ".so.2";
+                fullName = "lib" + name + ".so.12";
```
Comment (Alexsandruss) [inline java/com/intel/daal/utils/LibUtils.java:143]: @xwu99, have you encountered similar problem with libtbbmalloc.so.2? If yes, there is probably more correct fix:
```suggestion
                if (name.contains("malloc")) {
                    fullName = "lib" + name + ".so.2";
                }
                else {
                    fullName = "lib" + name + ".so.12";
                }
```

**Comment by xwu99** [comment]
@Alexsandruss Could you check CI error and help to fix this?

**Review by Alexsandruss — APPROVED** [review]
There are no CI errors connected to PR changes

**Pull request merged**